features/netsocket/NetworkInterface.h

```diff
@@ -31,6 +32,11 @@ class NetworkStack;
  */
 class NetworkInterface {
 public:
+
+    enum ConnectionStatusType {
+        down, local_up, global_up, undefined
```

Comment:
This should start uppercase `Down`, etc, each of them on the new line, good if with doxygen description for each status type.